spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ViewCFG.java

```diff
@@ -0,0 +1,176 @@
+
+public class ViewCFG implements Detector {
+
+    private final BugReporter bugReporter;
+    private Path tempDir;
+
+    public ViewCFG(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+        try {
+            tempDir = Files.createTempDirectory("cfg-");
+        } catch (IOException e) {
+            bugReporter.logError("Could not create temporary directory", e);
+        }
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (tempDir == null) {
+            return;
+        }
+        //
+        JavaClass cls = classContext.getJavaClass();
+        String classDirName = (cls.getPackageName() != "") ? (cls.getPackageName() + "." + cls.getClassName()) : cls.getClassName();
+        Path classDir;
+        //
+        try {
+            classDir = Files.createDirectory(Paths.get(tempDir.toString(), classDirName));
+        } catch (IOException e) {
+            bugReporter.logError("Could not create directory for class " + cls.getClassName(), e);
+            return;
+        }
+        //
+        for (Method method : cls.getMethods()) {
+            try {
+                analyzeMethod(classContext, method, classDir);
+            } catch (CFGBuilderException e) {
+                bugReporter.logError("Error analyzing method", e);
+            }
+        }
+    }
+
+    private void analyzeMethod(ClassContext classContext, Method method, Path classDir) throws CFGBuilderException {
+        Path methodFile = getMethodFile(classDir, method.getName());
+
+        try (PrintStream out = new PrintStream(Files.createFile(methodFile).toFile(), Charset.defaultCharset().name())) {
+            CFG cfg = classContext.getCFG(method);
+            out.println("digraph " + method.getName() + " {");
+            for (Iterator<BasicBlock> bi = cfg.blockIterator(); bi.hasNext();) {
+                BasicBlock block = bi.next();
+                if (block == cfg.getEntry()) {
+                    out.println("  Node" + block.getLabel() + " [shape=record label=\"{" + block.getLabel() +
+                            " (ENTRY) }\"];");
+                    continue;
+                }
+                //
+                if (block == cfg.getExit()) {
+                    out.println("  Node" + block.getLabel() + " [shape=record label=\"{" + block.getLabel() +
+                            " (EXIT) }\"];");
+                    continue;
+                }
+                //
+                out.print("  Node" + block.getLabel() + " [shape=record label=\"{" + block.getLabel());
+                if (block.getFirstInstruction() != null) {
+                    out.print("|");
+                }
+                for (Iterator<InstructionHandle> ii = block.instructionIterator(); ii.hasNext();) {
+                    InstructionHandle ins = ii.next();
+                    String insStr = ins.toString(false).replaceAll(" ->", "").replaceAll(" (\\d+)$", " #$1");
```

Comment:
The [`replaceAll()` method compiles RegExp internally](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#replaceAll-java.lang.String-java.lang.String-), and it make the performance bad.
So when we invoke this method in a loop, it is better to compile RegExp pattern in advance, and store it into a class field.